python/pyarrow/tests/conftest.py

```diff
@@ -311,3 +320,23 @@ def s3_server(s3_connection):
         finally:
             if proc is not None:
                 proc.kill()
+
+
+@pytest.fixture(scope='session')
+def gcs_server():
+    port = find_free_port()
+    env = os.environ.copy()
+    args = [sys.executable, '-m', 'testbench', '--port', str(port)]
```

Comment:
So basically this doesn't work if the GCS testbench was installed in a different Python.
Ideally the GCS testbench would export an executable script to avoid such issues (@coryan what do you think?).
In the meantime, we can take a `PYARROW_GCS_TESTBENCH_EXE` environment variable that points to the right Python executable and fall back on `sys.executable`.

Reply:
it actually seems a little stranger then that, I tried exporting the variable in various places and appending it to ~/.profile in various places and couldn't get it to stick.  I've cleaned up the approach to only install the file if the script exists (needs to see if this passes mac CI as well)

Reply:
Yeah, so the problem is that `conda-cpp.dockerfile` installs the GCS testbench with the Python version from the default conda install, but `conda-python.dockerfile` switches to another Python version.